File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/styles/CustomStyleInformation.java
package com.dexterous.flutterlocalnotifications.models.styles;

import androidx.annotation.Keep;

import java.util.Map;

@Keep
public class CustomStyleInformation extends DefaultStyleInformation {
    public String customContentView;
    public String customBigContentView;
    public Map<String, String> viewTexts;

    public CustomStyleInformation(
            String customContentView,
            String customBigContentView,
            Map<String, String> viewTexts) {
        super(false, false);
        this.customContentView = customContentView;
        this.customBigContentView = customBigContentView;
        this.viewTexts = viewTexts;
    }
}

File: flutter_local_notifications/lib/src/platform_specifics/android/styles/custom_style_information.dart
import 'style_information.dart';
import 'default_style_information.dart';

/// The Custom Android notification style.
class CustomStyleInformation extends DefaultStyleInformation {
    /// Constructs an instance of [CustomStyleInformation].
    const CustomStyleInformation({
        required this.customContentView,
        this.customBigContentView,
        this.viewTexts = const {}
    }): super(false, false);

    final String customContentView;

    final String? customBigContentView;

    final Map<String, String> viewTexts;
}